functions.opengraph.php

```diff
@@ -184,50 +198,64 @@ function jetpack_og_get_image( $width = 200, $height = 200, $max_images = 4 ) {
 		$author = get_queried_object();
 		if ( function_exists( 'get_avatar_url' ) ) {
 			// Prefer the core function get_avatar_url() if available, WP 4.2+
-			$image = get_avatar_url( $author->user_email, array( 'size' => $width ) );
+			$image['src'] = get_avatar_url( $author->user_email, array( 'size' => $width ) );
 		}
 		else {
 			$has_filter = has_filter( 'pre_option_show_avatars', '__return_true' );
-			if ( !$has_filter ) {
+			if ( ! $has_filter ) {
 				add_filter( 'pre_option_show_avatars', '__return_true' );
 			}
 			$avatar = get_avatar( $author->user_email, $width );
-			if ( !$has_filter ) {
+			if ( ! $has_filter ) {
 				remove_filter( 'pre_option_show_avatars', '__return_true' );
 			}
 
-			if ( !empty( $avatar ) && !is_wp_error( $avatar ) ) {
+			if ( ! empty( $avatar ) && ! is_wp_error( $avatar ) ) {
 				if ( preg_match( '/src=["\']([^"\']+)["\']/', $avatar, $matches ) );
-					$image = wp_specialchars_decode( $matches[1], ENT_QUOTES );
+					$image['src'] = wp_specialchars_decode( $matches[1], ENT_QUOTES );
 			}
 		}
 	}
 
-	if ( empty( $image ) )
+	if ( empty( $image ) ) {
 		$image = array();
-	else if ( !is_array( $image ) )
-		$image = array( $image );
+	} else if ( ! is_array( $image ) ) {
+		$image = array(
+			'src' => $image
+		);
+	}
 
 	// First fall back, blavatar
 	if ( empty( $image ) && function_exists( 'blavatar_domain' ) ) {
 		$blavatar_domain = blavatar_domain( site_url() );
-		if ( blavatar_exists( $blavatar_domain ) )
-			$image[] = blavatar_url( $blavatar_domain, 'img', $width, false, true );
+		if ( blavatar_exists( $blavatar_domain ) ) {
+			$image['src']    = blavatar_url( $blavatar_domain, 'img', $width, false, true );
+			$image['width']  = $width;
+			$image['height'] = $height;
+		}
 	}
 
 	// Second fall back, Site Logo
 	if ( empty( $image ) && ( function_exists( 'jetpack_has_site_logo' ) && jetpack_has_site_logo() ) ) {
-		$image[] = jetpack_get_site_logo( 'url' );
+		$image['src']        = jetpack_get_site_logo( 'url' );
+
+		$image_dimensions    = jetpack_get_site_logo_dimensions();
```

Comment:
We would need to find a place to add `jetpack_get_site_logo_dimensions()` in wpcom